sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-active-supplier-not-found-processor.ts

```diff
@@ -0,0 +1,118 @@
+ * Process a student that was not found on CAS.
+ */
+@Injectable()
+export class CASActiveSupplierNotFoundProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Create the new supplier and site on CAS using the student information.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    auth: CASAuthDetails,
+    summary: ProcessSummary,
+  ): Promise<ProcessorResult> {
+    if (evaluationResult.status !== CASEvaluationStatus.NotFound) {
+      throw new Error("Incorrect CAS evaluation result processor selected.");
+    }
+    summary.info(
+      `No active CAS supplier found. Reason: ${evaluationResult.reason}.`,
+    );
+    let result: CreateSupplierAndSiteResponse;
+    try {
+      const address = casSupplier.student.contactInfo.address;
+      result = await this.casService.createSupplierAndSite(auth.access_token, {
+        firstName: casSupplier.student.user.firstName,
+        lastName: casSupplier.student.user.lastName,
+        sin: casSupplier.student.sinValidation.sin,
+        emailAddress: casSupplier.student.user.email,
+        supplierSite: {
+          addressLine1: address.addressLine1,
+          city: address.city,
+          provinceCode: address.provinceState,
+          postalCode: address.postalCode,
+        },
+      });
+      summary.info("Created supplier and site on CAS.");
+    } catch (error: unknown) {
+      summary.error("Error while creating supplier and site on CAS.", error);
+      return { isSupplierUpdated: false };
+    }
+    try {
+      const [submittedAddress] = result.submittedData.SupplierAddress;
+      const now = new Date();
+      const systemUser = this.systemUsersService.systemUser;
+      const updateResult = await this.casSupplierRepo.update(
+        {
+          id: casSupplier.id,
+        },
+        {
+          supplierNumber: result.response.supplierNumber,
+          supplierName: result.submittedData.SupplierName,
+          status: "ACTIVE",
+          lastUpdated: now,
+          supplierAddress: {
+            supplierSiteCode: result.response.supplierSiteCode,
+            addressLine1: submittedAddress.AddressLine1,
+            city: submittedAddress.City,
+            provinceState: submittedAddress.Province,
+            country: submittedAddress.Country,
+            postalCode: submittedAddress.PostalCode,
+            status: "ACTIVE",
+            lastUpdated: now,
+          },
```

Comment:
what about `supplierProtected` and `siteProtected` in the update?

Reply:
Do you mean setting them explicitly to `null`?
Since the supplier was not found, it will be a new supplier and site. These properties should never exist or be created for a new supplier and site.

Reply:
These properties should never exist or be created for a new supplier and site. - Ah, is it? I was under the impression that it is set during new supplier and site creation.

Reply:
They are legacy properties. As far as I know, they only exist due to some SFAS requirement.
Either way, we do not have any direction about setting them.